### ISSUE TITLE
Initialise state when getInitialState() is added after the initial render using HMR

### ISSUE THREAD

**Issue by insin** [gaearon/react-proxy#38]
When I'm working with hot reloading and adding state to a `React.createClass` component which didn't have it before, I usually have to hack in a one-time call to `setState(this.getInitialState())` somewhere to initialise the state and avoid `TypeError: this.state is null` in `render()`.

Is it possible to detect that a `getInitialState()` has just been added to a component (or that state is `null` and `getInitialState()` exists, or some other heuristic) and have react-proxy do the `setState(this.getInitialState())` bit for me, and is react-proxy the right place to do this?

**Comment by vramana** [comment]
I think this is related to #15

**Comment by gaearon** [comment]
Yeah it's probably the right place. Feel free to try to PR this :-)
